maintenance/troubleshoot/commands.md

```diff
@@ -0,0 +1,499 @@
+
+```bash
+kubectl get pod -n kube-system -o wide
+```
+But if other pods fail, it is likely a different issue. Perform normal Kubernetes troubleshooting. For example:
+
+```bash
+kubectl describe pod kube-scheduler-ip-10-0-1-20.eu-west-1.compute.internal -n kube-system | tail -15
+``` 
+
+### Calico components
+
+#### View cluster CNI, IPAM, and policy 
+
+```bash
+cat /etc/cni/net.d/10-calico.conflist
+```
+
+#### Verify calicoctl matches cluster
+
+The cluster version and type must match the calicoctl version.
+
+```bash
+calicoctl version
+```
+
+For syntax:
+
+```bash
+calicoctl version -help
+```
+
+#### Check tigera operator status
+
+```bash
+kubectl get tigerastatus
+```
+
+```
+NAME     AVAILABLE   PROGRESSING   DEGRADED   SINCE
+calico   True        False         False      27h
+```
+
+#### Get tigera operator details
+
+```bash
+kubectl get tigerastatus
+```
+
+```
+NAME     AVAILABLE   PROGRESSING   DEGRADED   SINCE
+calico   True        False         False      27h
+```
+
+#### Verify component availability
+
+```bash
+kubectl get pod -n tigera-operator
+```
+
+#### View calico nodes
+
+```bash
+kubectl get pod -n calico-system -o wide
+```
+
+#### View cluster information
+
+```bash
+kubectl get installation -o yaml
+```
+
+```yaml
+apiVersion: v1
+items:
+- apiVersion: operator.tigera.io/v1
+  kind: Installation
+  metadata:
+    - apiVersion: operator.tigera.io/v1
+ spec:
+    calicoNetwork:
+      bgp: Enabled
+      hostPorts: Enabled
+      ipPools:
+      - blockSize: 26
+        cidr: 10.48.0.0/16
+        encapsulation: VXLANCrossSubnet
+        natOutgoing: Enabled
+        nodeSelector: all()
+      multiInterfaceMode: None
+      nodeAddressAutodetectionV4:
+        firstFound: true
+    cni:
+      ipam:
+        type: Calico
+      type: Calico
+```
+
+#### Run commands across multiple nodes
+
+Export THE_COMMAND_TO_RUN=date && for calinode in:
+
+```bash
+kubectl get pod -o wide -n calico-system | grep calico-node | awk '{print $1}'`; do echo $calinode; echo "-----"; 
+```
+
+```bash
+kubectl exec -n calico-system $calinode -- $THE_COMMAND_TO_RUN; printf "\n"; done calico-node-8xfmx
+```
+
+```
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-9t8s7
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-9cjhw
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-cb7ff
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-qoxvw
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+calico-node-wm5m2
+-----
+Defaulted container "calico-node" out of: calico-node, flexvol-driver (init), install-cni (init)
+```
+
+#### View pod info of pod that is not running
```

Comment:
this isn't specific to non-running pods, it may also be relevant for running pods.

Reply:
Fixed.